### PR TITLE
Add creation of delivery.txt

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR adds the creation of the `delivery.txt` file in the demultiplexed flow cell directory in the new `demultiplex finish` command. It seems like this file is necessary for the system d tasks to start delivering the data (?)

### Added
- Creation of `delivery.txt` file in the new `cg demultiplex finish` command

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by henrikstranneheim — COMMENTED** [review]
@karlnyr @Vince-janv @seallard  I don't remember why this flag file was needed? Do we still need it? Does it mark demux as complete?

**Comment by henrikstranneheim** [comment]
Then let's do this now and save that refactoring for later 👍

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by seallard** [comment]
Created an issue to track it https://github.com/Clinical-Genomics/cg/issues/2173

File: cg/meta/demultiplex/demux_post_processing.py [hunk 1]
```diff
@@ -136,6 +136,10 @@ def finish_flow_cell_temp(self, flow_cell_name: str) -> None:
         )
 
         self.update_sample_read_counts(sample_internal_ids=sample_ids)
+        self.create_delivery_file_in_flow_cell_directory(flow_cell_directory=flow_cell_dir)
+
+    def create_delivery_file_in_flow_cell_directory(self, flow_cell_directory: Path) -> None:
```
[the remark this answers is not in the document]
Comment (seallard) [reply cg/meta/demultiplex/demux_post_processing.py:141]: Just [above](https://github.com/Clinical-Genomics/cg/pull/2159/files#diff-afa9660521ae4a7f3be5393c339ace5b0741915c3b2a2c7f2bd04cf788d3ea26R139)

**Comment by seallard** [comment]
Deployed to stage and tested.

**Pull request merged**